clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2035,11 +2155,11 @@ void Sema::CheckSYCLKernelCall(FunctionDecl *KernelFunc, SourceRange CallLoc,
   }
 
   SyclKernelFieldChecker Checker(*this);
-
+  SyclKernelUnionBodyChecker Checker1(*this);
```

Comment:
Don't call this Checker1, probably call the one above "FieldChecker" and this one "UnionChecker".

Reply:
You ALSO  need to add the validity check of the UnionChecker to 2164.

Reply:
Done